### PR TITLE
[ShaderGraph] [2021.2] Fix for GPU instancing

### PR DESCRIPTION
**Pull request by cdxntchou**
### Purpose of this PR

Fix for GPU instancing issues
https://fogbugz.unity3d.com/f/cases/1318769/
   GPU instancing not working on custom properties, because we rename the instanced properties (appending _Array)

https://fogbugz.unity3d.com/f/cases/1338695/
   HDRP Target bug, gpu-instanced variant does not compile, caused by instanced prop macro being defined after the include that it was supposed to affect.

https://fogbugz.unity3d.com/f/cases/1319655/
   Same HDRP bug, but caused unintentionally by not exposing a property, causing it to fall out of SRP Batcher and back to old renderloop (with gpu instancing).

2021.2: https://github.com/Unity-Technologies/Graphics/pull/3734
2021.1: https://github.com/Unity-Technologies/Graphics/pull/4614/
2020.3: https://github.com/Unity-Technologies/Graphics/pull/4823

---
### Testing status
Describe what manual/automated tests were performed for this PR

Tested the instanceable property types, verified generated code was correct:
- [x] Float Properties
- [x] Vector 2/3/4 Properties
- [x] Matrix Properties
- [x] Color Properties
- [x] Diffusion Profile Properties

Tested rendering paths:
- [x] URP - Instanced, MPB, exposed -- uses GPU Instancing (old render loop)
- [x] URP - Instanced, !MPB, exposed -- uses SRP Batcher
- [x] URP - !Instanced, MPB, exposed -- uses individual draw calls (old render loop)
- [x] URP - !Instanced, !MPB, exposed -- uses SRP Batcher
- [x] URP - Instanced, MPB, !exposed -- uses GPU Instancing (old render loop)
- [x] URP - Instanced, !MPB, !exposed -- uses GPU Instancing (old render loop)
- [x] URP - !Instanced, MPB, !exposed -- uses individual draw calls (old render loop)
- [x] URP - !Instanced, !MPB, !exposed -- uses individual draw calls (old render loop)
- [x] HDRP - Instanced, MPB, exposed -- uses GPU Instancing (old render loop)
- [x] HDRP - Instanced, !MPB, exposed -- uses SRP Batcher
- [x] HDRP - !Instanced, MPB, exposed -- uses individual draw calls (old render loop)
- [x] HDRP - !Instanced, !MPB, exposed -- uses SRP Batcher
- [x] HDRP - Instanced, MPB, !exposed -- uses GPU Instancing (old render loop)
- [x] HDRP - Instanced, !MPB, !exposed -- uses GPU Instancing (old render loop)
- [x] HDRP - !Instanced, MPB, !exposed -- uses individual draw calls (old render loop)
- [x] HDRP - !Instanced, !MPB, !exposed -- uses individual draw calls (old render loop)

Tested hybrid DOTS rendering:
- [x] HDRP - Test Hybrid V1 instanced rendering with DOTS

Tested VFX integration:
- [x] HDRP - Test VFX integration with non-instanced (material) properties, HDRP Target path
- [x] HDRP - Test VFX integration with instanced properties, HDRP Target path
- [x] HDRP - Test VFX integration with global properties, HDRP Target path
- [x] HDRP - Test VFX integration with non-exposed properties, HDRP Target path (all black)
- [x] HDRP - Test VFX integration with non-instanced (material) properties, VFX Target path
- [x] HDRP - Test VFX integration with instanced properties, VFX Target path
- [x] HDRP - Test VFX integration with global properties, VFX Target path
- [x] HDRP - Test VFX integration with non-exposed properties, VFX Target path (all black)
- [x] URP - Test VFX integration with non-instanced (material) properties, VFX Target path
- [x] URP - Test VFX integration with instanced properties, VFX Target path
- [x] URP - Test VFX integration with global properties, VFX Target path
- [x] URP - Test VFX integration with non-exposed properties, VFX Target path (all black)
image

Added HDRP and URP Instancing Tests to test all combinations of "Exposed/Unexposed instanced property", "Material GPU instancing enabled/disabled", and "Material Property Block present/not present".

Before:  (compile failures show up purple)
image

After:
image

Yamato:

ShaderGraph Nightly Job: 🟢
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg%252Ffix%252Fgpu-instancing/.yamato%252Fall-shadergraph.yml%2523Nightly_ShaderGraph_trunk/7128151/job/pipeline

Universal PR Job: 🟡 -- failures match equivalent master branch
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg%252Ffix%252Fgpu-instancing/.yamato%252Fall-universal_split.yml%2523PR_Universal_Split_trunk/7128169/job/pipeline

failed job: URPUpdate_Top_Asset_Store on Win_DX11_playmode_mono_Linear on version trunk 

master equivalent: a7d5e885f4df8733b7323c2afa50e924cc53644b
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg%252Fold-master/.yamato%252Fall-universal_split.yml%2523PR_Universal_Split_trunk/7102788/job/pipeline

HDRP PR Job: 🟢
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg%252Ffix%252Fgpu-instancing/.yamato%252Fall-hdrp.yml%2523PR_HDRP_trunk/7106656/job/pipeline

---
### Comments to reviewers

This PR only modifies Universal to fix support for the (old) VFX Target path by adding the new UNITY_ACCESS_HYBRID_INSTANCED_PROP to VFXDefines.hlsl

### CODE REVIEW

**Review by jessebarker — APPROVED** [review]
LGTM

**Review by cinight — APPROVED** [review]
Manually checked HybridURPSamples, no regression compared to master. Also 2021.1 port has green on hybrid tests.

**Review by julienf-unity — APPROVED** [review]

**Review by pbbastian — APPROVED** [review]

**Review by erikabar — APPROVED** [review]
Looks good

**Review by GrantLamb-Unity — APPROVED** [review]
Good work on the testing + creation of the automated test!

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 1]
```diff
@@ -237,7 +237,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed support of Distortion with MSAA
 - Fixed contact shadow debug views not displaying correctly upon resizing of view.
 - Fixed an error when deleting the 3D Texture mask of a local volumetric fog volume (case 1339330).
-- Fixed some aliasing ussues with the volumetric clouds.
```
Comment (marctem) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md:240]: Looks like the minus sign got deleted here somehow? I don't want to break HD's changelog. And while you're at it, you could spell "issues" correctly. =)

Comment (cdxntchou) [reply com.unity.render-pipelines.high-definition/CHANGELOG.md:240]: Done!

**Review by marctem — APPROVED** [review]

**Pull request merged**